tensorflow_lite_support/python/task/text/nl_classifier.py

```diff
@@ -71,12 +73,25 @@ def create_from_options(cls, options: NLClassifierOptions) -> "NLClassifier":
       `NLClassifier` object that's created from `options`.
     Raises:
       ValueError: If failed to create `NLClassifier` object from
-        `NLClassifierOptions` such as missing the model.
+        `NLClassifierOptions` such as missing the model or if any of the
+        classification options is invalid.
       RuntimeError: If other types of error occurred.
     """
+    classification_options = options.classification_options
+
+    if classification_options.max_results == 0:
+      raise ValueError("Invalid `max_results` option: value must be != 0")
+
+    if classification_options.category_name_allowlist is not None and \
+        classification_options.category_name_denylist is not None:
+      if len(classification_options.category_name_allowlist) > 0 and \
+          len(classification_options.category_name_denylist) > 0:
+        raise ValueError(
+          "`class_name_allowlist` and `class_name_denylist` are mutually "
+          "exclusive options.")
+
     classifier = _CppNLClassifier.create_from_options(
```

Comment:
You should update the _CppNLClassifier to take only the BaseOptions.

Reply:
Updated.